server.js

```diff
@@ -78,6 +79,16 @@ if (config.startServer) {
         logger.addFileLogging(config.logFilename);
         logger.verbose('activated file logging: ' + config.logFilename);
     }
+
+    /* This synchronous code may seem gross, but we really do want to block
+       the below from running until we are able to grab config values from
+       AWS. */
+    try {
+        deasync(util.callbackify(awsHelper.init))();
+        deasync(util.callbackify(awsHelper.loadConfigSecrets))();
```

Comment:
I appreciate the minimally-invasive code change.

As an alternative, could we put all other top-level code below this into a function (placeholder name `topLevelInit()`) and re-arrange the `async.series()` call at the end of `server.js` to do something like:
```
const app = express();
module.exports.app = app;

if (config.startServer) {
    async.series([
        async () => {
            await awsHelper.init();
            await awsHelper.loadConfigSecrets();
        },
        async () => {
            topLevelInit();
        },
        ... all the other existing async.series() blocks
    ]);
}
```

We would need to call `server.topLevelInit()` in `tests/helperServer.js`.

Reply:
There's also https://itnext.io/node-js-script-writers-top-level-async-await-now-available-a1e86566d5f6 (we may have to update node, though).

I like the idea of cleaning up the main server file, and while we're at it we could change out all the sync code to be async/callback.